samples/SampleApp/Startup.cs

```diff
@@ -24,8 +26,24 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerFactory)
         {
             var logger = loggerFactory.CreateLogger("Default");
 
+            // Add an exception handler that prevents throwing due to large request body size
+            app.Use(async (context, next) =>
+            {
+                // Limit the request body to 1kb
+                context.Features.Get<IHttpMaxRequestBodySizeFeature>().MaxRequestBodySize = 1024;
+
+                try
+                {
+                    await next.Invoke();
+                }
+                catch (BadHttpRequestException ex) when (ex.StatusCode == StatusCodes.Status413RequestEntityTooLarge) { }
```

Comment:
Added a sample illustrating the use of the exposed status code. Before this change, this would not be possible and you'll see the following message in the log when the limit is exceeded:
```
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 POST http://localhost:5000/  1536
info: Microsoft.AspNetCore.Server.Kestrel[17]
      Connection id "0HLHB2D9PTLGQ" bad request data: "Request body too large."
Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException: Request body too large.
   at Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException.Throw(RequestRejectionReason reason) in C:\gh\KestrelHttpServer\src\Kestrel.Core\BadHttpRequestException.cs:line 41
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.ForContentLength.OnReadStarting() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\Http1MessageBody.cs:line 405
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.TryInit() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 154
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.CopyToAsync(Stream destination, CancellationToken cancellationToken) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 82
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpRequestStream.CopyToAsyncInternal(Stream destination, CancellationToken cancellationToken) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpRequestStream.cs:line 157
   at SampleApp.Startup.<>c__DisplayClass0_0.<<Configure>b__1>d.MoveNext() in C:\gh\KestrelHttpServer\samples\SampleApp\Startup.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at SampleApp.Startup.<>c.<<Configure>b__0_0>d.MoveNext() in C:\gh\KestrelHttpServer\samples\SampleApp\Startup.cs:line 35
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpProtocol.cs:line 530
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HLHB2D9PTLGQ", Request id "0HLHB2D9PTLGQ:00000001": An unhandled exception was thrown by the application.
Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException: Request body too large.
   at Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException.Throw(RequestRejectionReason reason) in C:\gh\KestrelHttpServer\src\Kestrel.Core\BadHttpRequestException.cs:line 41
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.ForContentLength.OnReadStarting() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\Http1MessageBody.cs:line 405
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.TryInit() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 154
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.CopyToAsync(Stream destination, CancellationToken cancellationToken) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 82
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpRequestStream.CopyToAsyncInternal(Stream destination, CancellationToken cancellationToken) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpRequestStream.cs:line 157
   at SampleApp.Startup.<>c__DisplayClass0_0.<<Configure>b__1>d.MoveNext() in C:\gh\KestrelHttpServer\samples\SampleApp\Startup.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at SampleApp.Startup.<>c.<<Configure>b__0_0>d.MoveNext() in C:\gh\KestrelHttpServer\samples\SampleApp\Startup.cs:line 35
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpProtocol.cs:line 530
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 463.6797ms 413
```

With the custom handler here, the exception is swallowed and the request will not show an error in the logs. However we'll still log the bad request message at an informational level:
```
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 POST http://localhost:5000/  1536
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 367.0046ms 200
info: Microsoft.AspNetCore.Server.Kestrel[17]
      Connection id "0HLHB2CH58PNL" bad request data: "Request body too large."
Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException: Request body too large.
   at Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException.Throw(RequestRejectionReason reason) in C:\gh\KestrelHttpServer\src\Kestrel.Core\BadHttpRequestException.cs:line 41
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1MessageBody.ForContentLength.OnReadStarting() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\Http1MessageBody.cs:line 405
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.TryInit() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 154
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.MessageBody.ConsumeAsync() in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\MessageBody.cs:line 132
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpProtocol.cs:line 602
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequestsAsync[TContext](IHttpApplication`1 application) in C:\gh\KestrelHttpServer\src\Kestrel.Core\Internal\Http\HttpProtocol.cs:line 447
```